kyo-core/shared/src/main/scala/kyo/scheduler/IOPromise.scala

```diff
@@ -128,7 +134,19 @@ private[kyo] class IOPromise[E, A](init: State[E, A]) extends Safepoint.Intercep
                             given Frame = Frame.internal
                             Log.unsafe.error("uncaught exception", ex)
         onCompleteLoop(this)
-    end onResult
+    end onComplete
+
+    inline def onInterrupt(inline f: Panic => Unit): Unit =
```

Comment:
Nice, this will be helpful for zio/cats integrations.

Reply:
yeah, I started working on them and saw that it'd be required otherwise we'd need that workaround like you did with the two promises